datajunction-clients/python/datajunction/client.py

```diff
@@ -180,6 +184,33 @@ def sql(  # pylint: disable=too-many-arguments
             return response.json()["sql"]
         return response.json()
 
+    #
+    # Get SQL for a node
+    #
+    def node_sql(  # pylint: disable=too-many-arguments
+        self,
+        node_name: str,
+        dimensions: Optional[List[str]] = None,
+        filters: Optional[List[str]] = None,
+        engine_name: Optional[str] = None,
+        engine_version: Optional[str] = None,
+    ):
+        """
+        Builds SQL for a node with the provided dimensions and filters.
+        """
+        response = self._session.get(
+            f"/sql/{node_name}",
+            params={
+                "dimensions": dimensions or [],
+                "filters": filters or [],
+                "engine_name": engine_name or self.engine_name,
+                "engine_version": engine_version or self.engine_version,
```

Comment:
Versions for engines are optional but this line would prevent us from selecting an engine where engine_version="" since `"" or self.engine_version` would always resolve to `self.engine_version`. I ran into this problem with similar logic in the query service and fixed it in the last PR (see [these](https://github.com/DataJunction/dj/blob/main/datajunction-query/djqs/api/tables.py#L37-L46) lines). I was trying to use engine_name of "default" and engine_version of "" and couldn't.

Reply:
Good catch! Fixing this now